R/format_WRS.R

```diff
@@ -0,0 +1,648 @@
+                                             .data$Species == "BT"  ~ species_codes[species_codes$SpeciesID == 14620,]$Species,
+                                             .data$Species == "FC"  ~ species_codes[species_codes$SpeciesID == 13490,]$Species,
+                                             .data$Species == "NUT" ~ species_codes[species_codes$SpeciesID == 14790,]$Species,
+                                             .data$Species == "CT"  ~ species_codes[species_codes$SpeciesID == 14610,]$Species,
+                                             .data$Species == "RS"  ~ species_codes[species_codes$SpeciesID == 11220,]$Species,
+                                             .data$Species == "SP"  ~ species_codes[species_codes$SpeciesID == 15980,]$Species)) %>%
+
+    dplyr::select(.data$BreedingSeason,
+                  .data$PopID,
+                  .data$Plot,
+                  .data$LocationID,
+                  .data$Species,
+                  .data$IndvID,
+                  .data$CaptureDate,
+                  .data$Mass,
+                  .data$Tarsus,
+                  .data$ChickAge,
+                  .data$ReleaseAlive,
+                  .data$UniqueBreedingEvent)
+
+  ## Read in primary data from adults
+  adult_data <- suppressWarnings(readxl::read_xlsx(path = paste0(db, "/WRS_PrimaryData.xlsx"),
+                                                   sheet = "Adults",
+                                                   col_types = "text")) %>%
+    janitor::clean_names(case = "upper_camel") %>%
+    janitor::remove_empty(which = "rows") %>%
+
+    ## Rename columns
+    dplyr::rename(BreedingSeason = .data$Year,
+                  Plot = .data$Site,
+                  LocationID = .data$NestboxId,
+                  IndvID = .data$RingId,
+                  ObserverID = .data$Obs,
+                  Sex_observed = .data$Sex,
+                  Age_observed = .data$Age,
+                  Mass = .data$Weight,
+                  WingLength = .data$WingLenght,
+                  Tarsus = .data$Tarsus) %>%
+
+    ## Handling different date formats in Excel
+    dplyr::mutate(CaptureDate = suppressWarnings(dplyr::case_when(grepl("/", .data$Date) ~ lubridate::dmy(.data$Date, quiet = TRUE),
+                                                                  TRUE ~ janitor::excel_numeric_to_date(as.numeric(.data$Date))))) %>%
+
+    ## Recode columns
+    ## TODO: Check about age codes
+    ## TODO: Check about aggression scoring - Should this be an experiment? Currently listed as OTHER
+    ## TODO: where() not namespaced
+    ## Check species codes
+    mutate(dplyr::across(where(is.character), ~dplyr::na_if(., "NA")),
+           PopID = "WRS",
+           BreedingSeason = as.integer(.data$BreedingSeason),
+           dplyr::across(c(.data$Mass, .data$WingLength, .data$Tarsus), ~ suppressWarnings(as.numeric(.x))),
+           CaptureTime = suppressWarnings(format(openxlsx::convertToDateTime(.data$Hour), "%H:%M")),
```

Comment:
This can be done with `janitor::convert_to_datetime()`. We should use this instead of introducing another dependency. I noticed that there are some times stored as numeric and some that are still strings e.g. row 841, so this will probably need some case_when statement to ensure these times aren't coerced to NA.

Reply:
I couldn't get convert_to_datetime to work, which is why I'd gone with the openxlsx function. I was able to find a base R solution that works though: `format(as.POSIXct(Sys.Date() + as.numeric(adult_data$Hour)), "%H:%M", tz="UTC"))`

This is used conditionally based on the format of the time in order to handle the fact that there is a mix of decimal days and character strings.